lib/multiplexer/wrappers.go

```diff
@@ -84,7 +84,16 @@ func (c *Conn) Detect() (Protocol, error) {
 
 // ReadProxyLine reads proxy-line from the connection.
 func (c *Conn) ReadProxyLine() (*ProxyLine, error) {
-	proxyLine, err := ReadProxyLine(c.reader)
+	var proxyLine *ProxyLine
+	protocol, err := c.Detect()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if protocol == ProtoProxyV2 {
```

Comment:
Do you think it's worth having a test to cover this? Is there anything else we could do to ensure that MySQL supports proxy protocol v2 / won't regress?

Reply:
I've added test for v2 for MySQL, Postgres, Mongo and Redis